core/src/main/java/com/amplifyframework/auth/AuthException.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.auth;
+
+import androidx.annotation.NonNull;
+
+import com.amplifyframework.AmplifyException;
+
+/**
+ * Exception thrown by Storage category plugins.
+ */
+public final class AuthException extends AmplifyException {
+
+    private static final long serialVersionUID = 1L;
+
+    /**
+     * Creates a new exception with a message, root cause, and recovery suggestion.
+     * @param message An error message describing why this exception was thrown
+     * @param throwable The underlying cause of this exception
+     * @param recoverySuggestion Text suggesting a way to recover from the error being described
+     */
+    public AuthException(
+            @NonNull final String message,
+            final Throwable throwable,
```

Comment:
Naming, `cause`

Reply:
We'd have to rename this for all the exceptions across the platform - do you think it's worth it?

Reply:
I do, yea. We should fix the other categories in a separate PR.